src/Validation/Validation.php

```diff
@@ -585,12 +590,41 @@ public static function datetime($check, $dateFormat = 'ymd', $regex = null)
         if (!empty($parts) && count($parts) > 1) {
             $date = rtrim(array_shift($parts), ',');
             $time = implode(' ', $parts);
+            if ($dateFormat === 'iso8601') {
+                $dateFormat = 'ymd';
+                $time = preg_split("/[TZ\-\+\.]/", $time);
+                $time = array_shift($time);
+            }
             $valid = static::date($date, $dateFormat, $regex) && static::time($time);
         }
 
         return $valid;
     }
 
+    /**
+     * Validates an iso8601 datetime format
+     * ISO8601 recognize datetime like 2019 as a valid date. To validate and check date integrity, use @see \Cake\Validation\Validation::datetime()
+     *
+     * @param string|\DateTimeInterface $check Value to check
+     *
+     * @return bool True if the value is valid, false otherwise
+     *
+     * @copyright Regex credits: https://www.myintervals.com/blog/2009/05/20/iso-8601-date-validation-that-doesnt-suck/
```

Comment:
This is not necessary. That blog post doesn't state any copyright license for the shown.

Reply:
Ok but that doesn't hurt. He did a good job and deserve it.

Reply:
`@see` is probably the normal link tag right?

Reply:
Ok, that's fair enough :)